pkg/build/api/validation/validation.go

```diff
@@ -241,6 +245,9 @@ func validateGitSource(git *buildapi.GitBuildSource) fielderrors.ValidationError
 	if len(git.HTTPSProxy) != 0 && !isValidURL(git.HTTPSProxy) {
 		allErrs = append(allErrs, fielderrors.NewFieldInvalid("httpsproxy", git.HTTPSProxy, "proxy is not a valid url"))
 	}
+	if strings.HasPrefix(git.URI, "git://") && hasProxy(git) {
+		allErrs = append(allErrs, fielderrors.NewFieldInvalid("uri", git.URI, "git:// protocol cannot be used with HTTP proxy"))
```

Comment:
ssh as well, right? Shouldn't we parse the source uri and require http or https scheme?